CNAME

```diff
@@ -0,0 +1 @@
+iree.dev
```

Comment:
Can you add a description to this PR with some context for this change?

Reply:
In particular, are you planning on fully renaming https://iree-org.github.io/iree/ to docs.iree.dev or iree.dev? We'll have a bunch of links to update in that case.

Reply:
I think both of these should still work: initially iree.dev and iree-org.github.io/iree/ should match with redirect from docs.iree.dev to iree.dev (github has single domain restriction, I could also flip it and have docs.iree.dev here and iree.dev be the redirect). I think full renaming would be nice though.

Reply:
iree.dev as the base, with docs.iree.dev redirecting there SGTM (at least until we have more subdomains and find something more fitting to sit at the unqualified root name)